src/utility/HCISharedMemTransport.cpp

```diff
@@ -338,8 +339,14 @@ static bool get_bd_address(uint8_t *bd_addr)
 
     bd_found = true;
   } else {
-    bd_addr = NULL;
-    bd_found = false;
+    uint8_t *otp_addr = OTP_Read(0);
+    if (otp_addr) {
+      memcpy(bd_addr, otp_addr, CONFIG_DATA_PUBADDR_LEN);
```

Comment:
In otp.h, otp structure is defined, maybe it should be better to use it?

With https://github.com/stm32duino/Arduino_Core_STM32/pull/1291 it has been renamed: `OTP_BT_t `

```suggestion
    OTP_BT_t *p_otp = OTP_Read(0);
    if (p_otp) {
      memcpy(bd_addr, p_otp->bd_address, CONFIG_DATA_PUBADDR_LEN);
```

Reply:
It depends on what the  type is, because OTP_Read returns a uint8_t *otp_addr

Reply:
This is the same used for HSE Tuning in the core so why differ here?
Main advantage using this is that if the OTP structure change (order for example), updating the structure is enough.

Reply:
done